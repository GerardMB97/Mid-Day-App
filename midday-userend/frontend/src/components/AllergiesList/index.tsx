import React from 'react';
import { View, Text, StyleSheet, FlatList } from 'react-native';

import Icon from 'react-native-vector-icons/Ionicons';

const styles = StyleSheet.create({
  listTitle: {
    position: 'relative',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    marginBottom: 20
  },
  titleText: {
    fontSize: 20
  },
  list: {
    height: 400,
    overflow: 'hidden',
    flexGrow: 0
  },
  closedArrow: {
    fontSize: 25,
    position: 'absolute',
    left: '100%',
    transform: [{ rotate: '90deg' }]
  },
  listItem: {
    borderBottomWidth: 1,
    borderBottomColor: 'black',
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  },
  allergyIcon: {
    fontSize: 25

  },
  allergyIconSelected: {
    color: 'red',
    fontSize: 25
  },
  selected: {
    color: 'red',
    borderBottomWidth: 1,
    borderBottomColor: 'black',
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center'
  }
});
export default function AllergiesList ({ ingredients }) {
  return (
    <View>
      <View style={styles.listTitle}>
        <Text style={styles.titleText}>Seleccione sus alergias</Text>
        <Icon style={styles.closedArrow} name="caret-forward-outline"></Icon>
      </View>
      <View style={styles.list}>
      <FlatList

        data ={ingredients}
        keyExtractor = {(item => item.category)}
        renderItem={({ item }) =>
        <View style={item.isAllergic ? styles.selected : styles.listItem}>
        <Text>{item.category}</Text>
        {item.isAllergic ? <Icon style = {styles.allergyIconSelected} name="sad-outline"></Icon> : <Icon style = {styles.allergyIcon} name="happy-outline"></Icon>}

        </View>

       }> </FlatList>
       </View>
    </View>

  );
}
