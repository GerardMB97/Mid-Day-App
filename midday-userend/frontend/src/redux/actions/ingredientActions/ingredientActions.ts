import axios from 'axios';
import ingredientActionTypes from './ingredientActionsTypes';
import { getIngredientsRoute } from '../../../constants/dataBase';
import { Dispatch } from 'redux';
import { User } from '../../../models';

export const getIngredients = (user:User) => {
  return async (dispatch:Dispatch) => {
    const { data } = await axios.get(getIngredientsRoute);

    dispatch({
      type: ingredientActionTypes.GET_INGREDIENTS,
      data,
      user
    });
  };
};
