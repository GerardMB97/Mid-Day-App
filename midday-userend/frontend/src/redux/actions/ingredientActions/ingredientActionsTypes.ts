const ingredientActionTypes = {
  GET_INGREDIENTS: 'GET_INGREDIENTS',
  UPDATE_ALLERGY: 'UPDATE_ALLERGY'
};

export default ingredientActionTypes;
