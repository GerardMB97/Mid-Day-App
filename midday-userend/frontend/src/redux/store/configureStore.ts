import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';
import rootReducer from '../reducers';
import initialState from './initialState';

function configureStore () {
  const composeEnhancers = compose;

  return createStore(
    rootReducer,
    initialState,
    composeEnhancers(applyMiddleware(reduxImmutableStateInvariant(), thunk))
  );
}

export default configureStore();
