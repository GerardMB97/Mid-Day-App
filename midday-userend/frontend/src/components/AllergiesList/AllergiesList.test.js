import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { fireEvent, render } from 'react-native-testing-library';
import AllergiesList from '.';
import * as actions from '../../redux/actions/ingredientActions/ingredientActions';

jest.mock('react-native-vector-icons/Ionicons', () => 'Icon');
jest.mock('../../redux/actions/ingredientActions/ingredientActions');

const mockStore = configureStore();
const store = mockStore({});
describe('Given an Allergies list component', () => {
  beforeEach(() => {
    jest.spyOn(actions, 'updateAllergies').mockReturnValueOnce({ type: '' });
  });
  describe('When rendered,', () => {
    test('Then it should match its snapshot', () => {
      const tree = render(<Provider store = {store}><AllergiesList ingredients = {[{ ingredient: 'carrot', isAllergic: false }, { ingredient: 'cucumber', isAllergic: true }]}></AllergiesList></Provider>);
      const button = tree.getByTestId('carrot');
      fireEvent.press(button);
      expect(tree).toMatchSnapshot();
    });
  });
  describe('When a button is clicked,', () => {
    test('Then it should invoke actions.updateAllergies', () => {
      const tree = render(<Provider store = {store}><AllergiesList ingredients = {[{ ingredient: 'carrot', isAllergic: false }, { ingredient: 'cucumber', isAllergic: true }]}></AllergiesList></Provider>);
      const button = tree.getByTestId('carrot');
      fireEvent.press(button);
      expect(actions.updateAllergies).toHaveBeenCalled();
    });
  });
});
