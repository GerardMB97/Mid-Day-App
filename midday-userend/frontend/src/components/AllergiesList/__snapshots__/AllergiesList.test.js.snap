// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Allergies list component When rendered, Then it should match its snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "borderBottomColor": "black",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "position": "relative",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
        }
      }
    >
      Seleccione sus alergias
    </Text>
    <Icon
      name="caret-forward-outline"
      style={
        Object {
          "fontSize": 25,
          "left": "100%",
          "position": "absolute",
          "transform": Array [
            Object {
              "rotate": "90deg",
            },
          ],
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flexGrow": 0,
        "height": 300,
        "overflow": "hidden",
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "ingredient": "carrot",
            "isAllergic": false,
          },
          Object {
            "ingredient": "cucumber",
            "isAllergic": true,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
            testID="carrot"
          >
            <Text>
              carrot
            </Text>
            <Icon
              name="happy-outline"
              style={
                Object {
                  "color": "green",
                  "fontSize": 25,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "color": "red",
                "flexDirection": "row",
                "height": 40,
                "justifyContent": "space-between",
              }
            }
            testID="cucumber"
          >
            <Text>
              cucumber
            </Text>
            <Icon
              name="sad-outline"
              style={
                Object {
                  "color": "red",
                  "fontSize": 25,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
